### PR TITLE
feat(permissions): created the scan incomplete warning bar

### PR DESCRIPTION
**Pull request by waabid**
#### Description of changes

This adds the scan incomplete warning bar component. The component is responsible for taking warning ids and translating them to message bars that can be displayed. The current design is such that each type of warning is only displayed once.

Note: the text for the iframe permission issue does not include any links at the moment but will in the future.

What the text looks like for the moment:

image

#### Pull request checklist

- [n/a] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). Check workflow guide at: `<rootDir>/docs/workflow.md`
- [x] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/DetailsView/components/scan-incomplete-warning.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { ScanIncompleteWarningId } from 'common/types/scan-incomplete-warnings';
+import { forOwn, isEmpty } from 'lodash';
+import { MessageBar, MessageBarType } from 'office-ui-fabric-react';
+import * as React from 'react';
+
+export type ScanIncompleteWarningProps = {
+    warnings: ScanIncompleteWarningId[];
+    onRender: () => void;
```
Comment (dbjorge) [inline src/DetailsView/components/scan-incomplete-warning.tsx]: What is this for? Presuming it's for the new onIframePermissionWarningShown, do we really want to be sending telemetry per-re-render? I'd expect it to be more appropriate to do that in response to scan complete actions with warnings present...

File: src/DetailsView/actions/details-view-action-message-creator.ts [hunk 2]
```diff
@@ -384,6 +384,11 @@ export class DetailsViewActionMessageCreator extends DevToolActionMessageCreator
         this.dispatcher.sendTelemetry(TelemetryEvents.DETAILS_VIEW_OPEN, telemetryData);
     }
 
+    public onIframePermissionWarningShown = () => {
```
Comment (dbjorge) [inline src/DetailsView/actions/details-view-action-message-creator.ts]: Would it be better to name this like the `ScanIncompleteWarning` pattern and include which warnings were present in the telemetryData?

File: src/DetailsView/components/scan-incomplete-warning.tsx (hunk 1 above)
Comment (waabid) [reply src/DetailsView/components/scan-incomplete-warning.tsx]: Yes, I was thinking about this and removing it as well; this pattern also doesn't work for other warnings that might be shown.

File: src/DetailsView/actions/details-view-action-message-creator.ts (hunk 2 above)
Comment (waabid) [reply src/DetailsView/actions/details-view-action-message-creator.ts]: I think I'm going to remove the onRender functionality and that likely means this is not necessary.

File: src/common/extension-telemetry-events.ts [hunk 3]
```diff
@@ -61,6 +61,7 @@ export const ALL_RULES_EXPANDED: string = 'allRulesExpanded';
 export const ALL_RULES_COLLAPSED: string = 'allRulesCollapsed';
 export const RESCAN_VISUALIZATION: string = 'rescanVisualization';
 export const EXISTING_TAB_URL_UPDATED: string = 'existingTabUrlUpdated';
+export const IFRAME_PERMISSION_WARNING_SHOWN: string = 'iframePermissionWarningShown';
```
Comment (dbjorge) [inline src/common/extension-telemetry-events.ts]: Intended to still be here?

**Pull request merged**